### PR TITLE
Misc unit test fixes

### PR DESCRIPTION
**Pull request by rickpasetto**
This fixes unit test failures that I regularly see locally, not sure why CI doesn't see it.  Maybe my laptop runs faster?   Or slower? 🤷‍♂️

### CODE REVIEW

File: LoopTests/ViewModels/BolusEntryViewModelTests.swift [hunk 1]
```diff
@@ -158,6 +158,7 @@ class BolusEntryViewModelTests: XCTestCase {
         bolusEntryViewModel.enteredManualGlucose = Self.exampleManualGlucoseQuantity
         XCTAssertEqual(Self.exampleBolusQuantity, bolusEntryViewModel.enteredBolus)
         waitOnMain()
+        waitOnMain()
```
Comment (darinkrauss) [inline LoopTests/ViewModels/BolusEntryViewModelTests.swift:163]: Why two calls to `waitOnMain`? Why not just use `waitOnMain(timeout: 2)`?

Comment (darinkrauss) [reply LoopTests/ViewModels/BolusEntryViewModelTests.swift:163]: Also applies to all other cases where `waitOnMain` is called twice in a row.

**Review by darinkrauss — APPROVED** [review]
I've never seen these errors in my local build. I suspect this has more to do with different versions of Xcode. Circle is using 12.2, my local build is using 12.3, and IIRC you are using the latest. IMHO, we (and Circle) should all be using the same version of Xcode.

I'll approve, but consider this and my comment below.

**Comment by rickpasetto** [comment]
You are correct, @darinkrauss ... these failures are almost certainly because of Xcode 12.5.  In fact, I tried this in CI, and they do [fail](https://app.circleci.com/pipelines/github/tidepool-org/LoopWorkspace/3012/workflows/0a1859df-d1e2-411e-ac46-faa56e70ce66).  So this is a problem just waiting to happen, so I'm trying to fix it before it blocks CI builds (and devs).

File: LoopTests/ViewModels/BolusEntryViewModelTests.swift (hunk 1 above)
Comment (rickpasetto) [reply LoopTests/ViewModels/BolusEntryViewModelTests.swift:163]: This is the biggest mystery to me.  I have tried increasing the `timeout`, but that doesn't help.  There's something about having to wait twice that is required to make these tests pass.  My other thought was that some other test isn't cleaning up properly, leaving some extra cruft in the `.main` queue, but that doesn't seem to be true, either.  What I will say is that `BolusEntryViewModel` plays some weird tricks with `Combine`, including `debounce`-ing (which I _thought_ was disabled for unit tests).  I could put a comment in here, but I have been unable to get to the bottom of it 🤷‍♂️ .  Any ideas?

Comment (darinkrauss) [reply LoopTests/ViewModels/BolusEntryViewModelTests.swift:163]: Nope, you've tried what I would have tried. Sounds good.

**Pull request merged**